web/webkit/src/main/scala/net/liftweb/http/LiftRulesGuardedSetting.scala

```diff
@@ -0,0 +1,126 @@
+package net.liftweb.http
+
+import net.liftweb.util.{HasCalcDefaultValue, LiftValue}
+
+object LiftRulesGuardedSetting {
+  type StackTrace = Array[StackTraceElement]
+
+  /**
+    * Base class for all possible violations which LiftRulesGuardedSetting warns you about
+    * @param settingName
+    * @param stackTrace
+    * @param message
+    */
+  abstract class SettingViolation(settingName: String, stackTrace: StackTrace, message: String) extends Serializable {
+    /**
+      * Converts this violation into an Exception which can be handed to a logger for clean message printing
+      * @return
```

Comment:
Alright, fine on both. This is where I kinda hate scaladoc because it wants me to add more words to my already descriptive names. LOL